signer/tests/integration/postgres.rs

```diff
@@ -1676,11 +1755,103 @@ async fn can_sign_deposit_tx_rejects_not_in_signer_set() {
+
+    let mut setup = TestSweepSetup::new_setup(&rpc, &faucet, 1_000_000, &mut rng);
+
+    let context_window = 20;
+
+    // Adding a block, we will use it to store the complete deposit event later
+    let chain_tip: BitcoinBlockHash = faucet.generate_blocks(1).pop().unwrap().into();
+
+    // We need to manually update the database with new bitcoin block
+    // headers.
+    crate::setup::backfill_bitcoin_blocks(&db, rpc, &chain_tip).await;
+
+    // This isn't technically required right now, but the deposit
+    // transaction is supposed to be there, so future versions of our query
+    // can rely on that fact.
+    setup.store_deposit_tx(&db).await;
+
+    // We take the sweep transaction as is from the test setup and
+    // store it in the database.
+    setup.store_sweep_tx(&db).await;
+
+    // The request needs to be added to the database. This stores
+    // `setup.deposit_request` into the database.
+    setup.store_deposit_request(&db).await;
+
+    // TODO: Create the initial transaction sweep package without any
+    // withdrawals and have a separate method for creating that sweep (since
+    // it's not realistic to have the withdrawal in the same sweep as the
+    // deposit). Then we wouldn't have to do this.
+    setup.store_withdrawal_request(&db).await;
+
+    // Store outstanding sweep transaction packages in the database.
+    setup.store_sweep_transactions(&db).await;
+
+    let stacks_tip = db
+        .get_stacks_chain_tip(&chain_tip.into())
+        .await
+        .unwrap()
+        .unwrap();
+
+    // First, let's check we get the deposit
+    let requests = db
+        .get_swept_deposit_requests(&chain_tip.into(), context_window)
+        .await
+        .unwrap();
+    assert_eq!(requests.len(), 1);
+
+    // Now we push the event to a stacks block anchored to the chain tip
+    let original_event_block = StacksBlock {
+        block_hash: fake::Faker.fake_with_rng(&mut rng),
+        block_height: stacks_tip.block_height + 1,
+        parent_hash: stacks_tip.block_hash,
+        bitcoin_anchor: chain_tip.into(),
+    };
+    db.write_stacks_block(&original_event_block).await.unwrap();
+
+    let event = CompletedDepositEvent {
+        txid: fake::Faker.fake_with_rng::<StacksTxId, _>(&mut rng).into(),
+        block_id: *original_event_block.block_hash,
+        amount: setup.deposit_request.amount,
+        outpoint: setup.deposit_request.outpoint,
+    };
+    db.write_completed_deposit_event(&event).await.unwrap();
+
+    // The deposit should be confirmed now
+    let requests = db
+        .get_swept_deposit_requests(&chain_tip.into(), context_window)
+        .await
+        .unwrap();
+
+    assert!(requests.is_empty());
+
+    // Now assume we have a reorg: the new bitcoin chain is `sweep_block_hash`
+    // and the complete deposit event is no longer in the canonical chain.
+    // The deposit should no longer be confirmed.
+    let requests = db
```

Comment:
Hm, I'm trying to reason whether or not this tests what the test description says... Doesn't this test that the Bitcoin chain has re-org'd? If i'm reading the description correctly, it reads to me that the Bitcoin chain is still canonical but that the `complete-deposit` transaction has been orphaned in Stacks (a Stacks re-org without Bitcoin re-org).

But maybe I'm reading things completely wrong?

Reply:
Hmm, I think the test description is only supposed to check that it the transaction has been affected by a reorg, it doesn't specify how. We may want more tests that check for Stacks reorgs. This one does seem bitcoin specific, but that is fine.

Reply:
The description says that the sweep tx has not been reorg'd but the `complete-deposit` has, which reads to me like a stacks reorg without bitcoin, but if this is fine with you then I'm fine :)

Reply:
Oh I see, yeah that interpretation makes sense. I think we just need to make sure that the bitcoin reorg doesn't affect the sweep transaction, which it shouldn't because we use the `sweep_block_hash` as the chain tip below.